### ISSUE TITLE
Rule proposal: spellcheck

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1442]
There is an [`eslint-plugin-spellcheck` plugin](https://github.com/aotaduy/eslint-plugin-spellcheck/), but it uses an old lib [`hunspell-spellchecker`](https://github.com/GitbookIO/hunspell-spellchecker), we can use [cspell](https://github.com/streetsidesoftware/cspell).

The idea is to reduce tooling costs,  WDYT? 

## Fail

```js
const unicorrn = 'unicorrn';
```

## Pass

```js
const unicorn = 'unicorn';
```

**Comment by sindresorhus** [comment]
Would be nice. The install size a bit big though: https://packagephobia.com/result?p=cspell Doesn't bother me, but some users might complain.

**Comment by fisker** [comment]
cspell already created one https://github.com/streetsidesoftware/cspell/pull/2379 👍

**Comment by sindresorhus** [comment]
This is probably best as a separate plugin. It's quite big and it would have a lot of flaws. I don't really want to maintain that.
